libsolidity/ast/ASTEnums.h

```diff
@@ -50,5 +51,22 @@ inline std::string stateMutabilityToString(StateMutability const& _stateMutabili
 	}
 }
 
+class Type;
+
+/// Container for function call parameter types & names
+struct FuncCallArguments
+{
+	/// Types of arguments
+	std::shared_ptr<std::vector<std::shared_ptr<Type const>>> types;
+	/// Names of the arguments if given, otherwise unset
+	std::shared_ptr<std::vector<ASTPointer<ASTString>>> names;
+
+	size_t numArguments() const { return types ? types->size() : 0; }
+	size_t numNames() const { return names ? names->size() : 0; }
+	bool hasNamedArguments() const { return names && names->empty(); }
```

Comment:
`!empty`?

Reply:
ohw. Indeed. I suppose it worked anyway because it would either be set or non-empty.

Reply:
hmm no that doesn't make sense. I'll investigate

Reply:
it worked anyway because the error was caught later in the argument mapping. Now that it is fixed, it bails out earlier with a fatal error